src/xtgeo/surface/regular_surface.py

```diff
@@ -2053,35 +2068,227 @@ def divide(self, other):
     # Interacion with points
     # ==================================================================================
 
-    def gridding(self, points, method="linear", coarsen=1):
-        """Grid a surface from points.
+    def gridding(
+        self,
+        input_data: Points | Polygons | "RegularSurface",
+        method: Literal[
+            "linear",
+            "cubic",
+            "nearest",
+            "inverse_distance",
+            "idw",
+            "radial_basis",
+            "rbf",
+            "moving_average",
+            "kriging",
+        ] = "linear",
+        coarsen: int = 1,
+        merge_close_points: float | None = None,
+        merge_method: Literal[
```

Comment:
This gridding method starts to get quite complicated. One simplification could be to remove these as arguments here, and just have a tip in the documentation that it is possible to run the merging of close points as a preprocess step on the point instance before. 

It kinda applies to the `extrapolation` option in the `method_option` as well - it could be run as a simple `surf.fill()` afterwards..

Reply:
You're right; will simplify!